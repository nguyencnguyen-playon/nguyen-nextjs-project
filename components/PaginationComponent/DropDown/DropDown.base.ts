import { css } from 'styled-components';

export const baseStyles = {
  inputWrapper: css`
    width: 60px;
    position: relative;
    border: 1px solid #ccc;
    border-radius: 2px;
    height: 40px;
  `,
  select: css<{ disabled?: boolean }>`
    width: 100%;
    border: none;
    background: #fff;
    font-family: Rubik-Regular;
    color: #323232;
    font-size: 14px;
    appearance: none;
  `,
  iconWrapper: css`
    position: absolute;
    top: 12px;
    right: 5%;
  `
};
