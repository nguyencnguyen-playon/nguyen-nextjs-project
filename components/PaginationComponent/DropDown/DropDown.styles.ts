import { css } from 'styled-components';
import { baseStyles } from './DropDown.base';

export const styles = {
  ...baseStyles,
  select: css<{ disabled?: boolean }>`
    ${baseStyles.select};
    height: 100%;
    width: 100%;
    padding-left: 25%;
    ${({ disabled }) => disabled && 'cursor: not-allowed'};
  `
};
