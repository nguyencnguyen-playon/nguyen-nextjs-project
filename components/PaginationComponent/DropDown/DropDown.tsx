import styled from 'styled-components';
import React from 'react';
import { styles } from './DropDown.styles';
import { DownOutlined } from '@ant-design/icons';

const DropdownContainer = styled.div`
  ${styles.inputWrapper}
`;

const DropdownSelect = styled.select`
  ${styles.select}
`;

const DropdownOption = styled.option``;

const IconWrapper = styled.div`
  ${styles.iconWrapper}
`;
interface DropdownProps {
  options: { label: string; value: string }[];
  selectedValue: string;
  onValueChange: (value: string) => void;
}

export const Dropdown: React.FC<DropdownProps> = ({ options, selectedValue, onValueChange }) => {
  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    onValueChange(event.target.value);
  };

  return (
    <DropdownContainer>
      <DropdownSelect value={selectedValue} onChange={handleSelectChange}>
        {options.map(({ value }) => (
          <DropdownOption key={value} value={value}>
            {value}
          </DropdownOption>
        ))}
      </DropdownSelect>
      <IconWrapper>
        <DownOutlined />
      </IconWrapper>
    </DropdownContainer>
  );
};
