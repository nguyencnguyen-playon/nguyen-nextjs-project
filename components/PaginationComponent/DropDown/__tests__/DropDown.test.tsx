import React from 'react';
import { render } from '@testing-library/react';
import { Dropdown } from '../DropDown';

const props = {
    options: [{ label: '10', value: '10' }, { label: '20', value: '20' }],
    selectedValue: '10',
    onValueChange: jest.fn()
};

describe('Test Dropdown Web component ', () => {
    test('should match its snapshot', async () => {
        const component = render(<Dropdown {...props} />);
        expect(component).toMatchSnapshot();
    });
});
