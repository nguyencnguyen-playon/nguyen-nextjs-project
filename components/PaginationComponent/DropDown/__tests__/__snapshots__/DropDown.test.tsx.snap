// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dropdown Web component  should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beqWNU gbqjiq"
      >
        <select
          class="sc-gueXAH fwRQqY"
        >
          <option
            class="sc-dmqGRC"
            value="10"
          >
            10
          </option>
          <option
            class="sc-dmqGRC"
            value="20"
          >
            20
          </option>
        </select>
        <div
          class="sc-hLseSp eewNXK"
        >
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beqWNU gbqjiq"
    >
      <select
        class="sc-gueXAH fwRQqY"
      >
        <option
          class="sc-dmqGRC"
          value="10"
        >
          10
        </option>
        <option
          class="sc-dmqGRC"
          value="20"
        >
          20
        </option>
      </select>
      <div
        class="sc-hLseSp eewNXK"
      >
        <span
          aria-label="down"
          class="anticon anticon-down"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
